File: src/main/rooms/Menu.java
package main.rooms;

import main.buttons.iconbuttons.IconButton;
import main.buttons.menubuttons.LoadGameButton;
import main.buttons.menubuttons.MenuButton;
import main.buttons.menubuttons.NewGameButton;
import main.buttons.menubuttons.PacksButton;
import processing.core.PConstants;
import processing.core.PFont;

public class Menu extends Room {

    private static PFont titleFont;

    private LoadGameButton loadGame;
    private NewGameButton newGame;
    private PacksButton packs;
    private MenuButton achievements;
    private IconButton settings;

    public Menu() {
        titleFont = main.createFont("resources/fonts/Alchemy Gold.ttf", 128);

        this.loadGame = new LoadGameButton();
        this.newGame = new NewGameButton();
        this.packs = new PacksButton();
        this.achievements = new MenuButton("achievements");
        this.settings = new IconButton("resources/images/settings_button.png") {
            @Override
            public void clicked() {
                main.switchRoom(main.settingsRoom);
            }
        };
    }

    @Override
    public void setup() {

    }

    @Override
    public void draw() {
        // draw objects that are specific to menu
        //main.image(background, 0, 0, main.screenWidth, main.screenHeight);
        main.fill(0);
        main.rect(0,0,main.screenWidth, main.screenHeight);
        main.textFont(titleFont, 220);
        main.textAlign(PConstants.CENTER, PConstants.TOP);
        main.fill(255);
        main.text(main.getLanguageSelected().getLocalizedString("menu", "alchemy"), main.screenWidth / 2F, 20);

        int y = 340;
        final int gap = 10;

        this.loadGame.draw(main.screenWidth / 2F - MenuButton.WIDTH / 2F, y);
        y += MenuButton.HEIGHT + gap;

        this.newGame.draw(main.screenWidth / 2F - MenuButton.WIDTH / 2F, y);
        y += MenuButton.HEIGHT + gap;

        this.packs.draw(main.screenWidth / 2F - MenuButton.WIDTH / 2F, y);
        y += MenuButton.HEIGHT + gap;

        this.achievements.draw(main.screenWidth / 2F - MenuButton.WIDTH / 2F, y);

        this.settings.draw(30, main.screenHeight - 30 - IconButton.SIZE);
    }

    @Override
    public void mousePressed() {
        this.loadGame.mousePressed();
        this.newGame.mousePressed();
        this.packs.mousePressed();
        this.achievements.mousePressed();
        this.settings.mousePressed();
    }

}

File: src/main/rooms/Menu.kt
package main.rooms

import main.Language
import main.Main
import main.buttons.iconbuttons.IconButton
import main.buttons.menubuttons.LoadGameButton
import main.buttons.menubuttons.MenuButton
import main.buttons.menubuttons.NewGameButton
import main.buttons.menubuttons.PacksButton
import processing.core.PConstants
import processing.core.PFont

object Menu : Room() {

    private val loadGame: LoadGameButton = LoadGameButton()
    private val newGame: NewGameButton = NewGameButton()
    private val packs: PacksButton = PacksButton()
    private val achievements: MenuButton = MenuButton("achievements")
    private val settings: IconButton
    private val titleFont: PFont = Main.createFont("resources/fonts/Alchemy Gold.ttf", 128f)

    init {
        settings = object : IconButton("resources/images/settings_button.png") {
            override fun clicked() {
                Main.switchRoom(SettingsRoom)
            }
        }
    }

    override fun setup() {}

    override fun draw() {
        Main.textFont(titleFont, 220f)
        Main.textAlign(PConstants.CENTER, PConstants.TOP)
        Main.fill(255)
        Main.text(Language.languageSelected.getLocalizedString("menu", "alchemy"), Main.screenWidth / 2f, 20f)
        var y = 340
        val gap = 10
        loadGame.draw(Main.screenWidth / 2f - MenuButton.WIDTH / 2f, y.toFloat())
        y += MenuButton.HEIGHT + gap
        newGame.draw(Main.screenWidth / 2f - MenuButton.WIDTH / 2f, y.toFloat())
        y += MenuButton.HEIGHT + gap
        packs.draw(Main.screenWidth / 2f - MenuButton.WIDTH / 2f, y.toFloat())
        y += MenuButton.HEIGHT + gap
        achievements.draw(Main.screenWidth / 2f - MenuButton.WIDTH / 2f, y.toFloat())
        settings.draw(30f, (Main.screenHeight - 30 - IconButton.SIZE).toFloat())
    }

    override fun mousePressed() {
        loadGame.mousePressed()
        newGame.mousePressed()
        packs.mousePressed()
        achievements.mousePressed()
        settings.mousePressed()
    }
}